estate/tests/test_estate_property.py

```diff
@@ -0,0 +1,55 @@
+from odoo.exceptions import UserError
+from odoo.tests import tagged
+from odoo.tests.common import Form, TransactionCase
+
+
+@tagged("post_install", "-at_install")
+class EstatePropertyTestCase(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.property = cls.env["estate.property"].create(
+            {
+                "name": "Property A",
+                "expected_price": 200000,
+                "state": "new",
+            }
+        )
+
+    def test_should_be_set_as_sold(self):
+        """When a property is set as sold, it should change its state to 'sold'"""
+
+        self.property.selling_price = 200000
+        self.property.state = "offer_accepted"
+        self.property.action_set_property_as_sold()
```

Comment:
@JesusValdez96 I am not able to create a _property offer_ unlinked to a property because the `property_id` is required.
The _property offer form view_ does not include the `property_id` field due it is only used within the _property form view_, and it is set automatically, so if I create the offer with `Form` class it raises this error: 
![image](https://github.com/gerashdo/odoo-gettingstarted/assets/55073086/b9504b90-7759-4083-a2da-7077c84a7b47)

![image](https://github.com/gerashdo/odoo-gettingstarted/assets/55073086/79f086ee-0894-4b0a-add1-1233f92b6b2e)

I could create an offer in `setUpClass` without using the `Form `class and use it to simulate the offer acceptance, if it seems correct to you.

Reply:
@gerashdo 
oh no, I meant that you add the offer to the property with the Form, the offer can be created without Form but linked to the property with it.

you said "I am not able to create a property offer unlinked to a property because the property_id is required" and "I could create an offer in setUpClass without using the Form" . Confirm if you can create that offer without Form and without property_id please.
If no, then use form with the property of setUpClass and add a new offer there.
with Form(self.property) as property:
    property.offer_ids=...